libraries/esp8266/examples/math/math.ino

```diff
@@ -0,0 +1,30 @@
+/*
+  Small math example, checking whether we can support various Arduino math operations.
```

Comment:
I strongly advise rewriting this as a device side test instead of an example. You can then add explicit checks for specific functionality.

Reply:
Ah. Meaning, the right place is `tests/device/test_sw_...`?
Only issue is (as far as I can understand the makefile MOCK=0 / MOCK=1), it gets picked up only by the host gcc and it will not be compiled by the xtensa toolchain too? The general issue was different code, so it needs to be able to be built by both.

Reply:
Device tests can always be compiled to run on device.
Some of them can also be compiled by both host and device compilers.
Current convention is to call them `test_sw_<name>`.

Reply:
Sure. Tried it out both with g++ 9.3.1 on host and running PR tree on device.
b/c gcc 10 pr is merged, it is less relevant to the testing, but it is still a useful patch

Reply:
No, CI currently does not run device tests.
But we could add device tests that can also be run in the test framework.

Reply:
As noted above, we don't really even need to run it, I am not really keen on it beeing a test at all. fwiw, I have tried a small example having *only* `static_assert(...condition..., "");`, because everything is already known at compile time.

Can you please point to the correct place, then? Only requirement is that something tries to build a .cpp with both xtensa and host's compiler.

Reply:
It's in `test/device`. There is for example `test_time` which may not run with emulation, and `test_sw_http_client` which runs on both (it was first called `test_http_client` then later renamed after proving to also work on host).